dht.cpp

```diff
@@ -64,7 +64,8 @@ void load_dht_state(BencEntity* ent)
     fseek(f, 0, SEEK_END);
     long fsize = ftell(f);
     fseek(f, 0, SEEK_SET);
-    char *buf = (char*)malloc(fsize);
+    char *buf = (char*)malloc(fsize + 1);
+    buf[fsize] = '\0'; // The BencEntity::Parse function expects this to be a null terminated string.
```

Comment:
While the dht.dat file was being read, the last function that was called to read it was [BencEntity::ParseNum](https://github.com/bittorrent/libbtutils/blob/d39069c55c463a1b5ff51dad04ac2f40de6e3060/src/bencoding.cpp#L544). Note that the function doesn't take as an argument the end boundary of the buffer and so if the string wasn't null terminated it was reading past that boundary.